### PR TITLE
Add `mc admin bucket remote edit` to edit credentials

### PR DESCRIPTION
**Pull request by poornas**
for existing bucket targets

### CODE REVIEW

**Comment by kerneltime** [comment]
Would it be better to have an update API within minio? vs read and then modify?

**Comment by poornas** [comment]
> Would it be better to have an update API within minio? vs read and then modify?

keeping it client side helps give actionable error messages for wrong user input

**Review by vadmeste — COMMENTED** [review]
Some comments, LGTM otherwise and tested

File: cmd/admin-bucket-remote-edit.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+		cli.ShowCommandHelpAndExit(ctx, ctx.Command.Name, 1) // last argument is exit code
+	}
+}
+
+// fetchRemoteEditTarget - returns the dest bucket, dest endpoint, access and secret key
+func fetchRemoteEditTarget(cli *cli.Context) (bktTarget *madmin.BucketTarget) {
+	args := cli.Args()
+	argCount := len(args)
+	if argCount != 2 {
+		fatalIf(probe.NewError(fmt.Errorf("Missing Remote target configuration")), "Unable to parse remote target")
```
[the remark this answers is not in the document]
Comment (poornas) [reply cmd/admin-bucket-remote-edit.go]: redundant check, removing

File: cmd/admin-bucket-remote-edit.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	}
+	_, sourceBucket := url2Alias(args[0])
+	TargetURL := args[1]
+	parts := targetKeys.FindStringSubmatch(TargetURL)
+	if len(parts) != 6 {
+		fatalIf(probe.NewError(fmt.Errorf("invalid url format")), "Malformed Remote target URL")
```
[the remark this answers is not in the document]
Comment (poornas) [reply cmd/admin-bucket-remote-edit.go:94]: These errors can be caught only while parsing args[1], not cli flag errors - so this seems to be the best place for this check.

File: cmd/admin-bucket-remote-edit.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	}
+	accessKey := parts[2]
+	secretKey := parts[3]
+	parsedURL := fmt.Sprintf("%s%s", parts[1], parts[4])
+	TargetBucket := strings.TrimSuffix(parts[5], slashSeperator)
+	TargetBucket = strings.TrimPrefix(TargetBucket, slashSeperator)
+	u, cerr := url.Parse(parsedURL)
+	if cerr != nil {
+		fatalIf(probe.NewError(cerr), "Malformed Remote target URL")
```
[the remark this answers is not in the document]
Comment (poornas) [reply cmd/admin-bucket-remote-edit.go:103]: same as above

**Review by vadmeste — APPROVED** [review]
LGTM & tested

**Comment by poornas** [comment]
@kerneltime, PTAL

**Review by kerneltime — APPROVED** [review]

**Pull request merged**